_posts/2020-11-26-torch-brain-segmentation/torch_brain_segmentation.Rmd

```diff
@@ -0,0 +1,705 @@
+
+Zooming in on images, we're not looking for a single label; instead, we want to classify every pixel according to some criterion:
+
+-   In medicine, we may want to distinguish between different cell types, or identify tumors.
+
+-   In various earth sciences, satellite data are used to segment terrestrial surfaces.
+
+-   To enable use of custom backgrounds, video-conferencing software has to be able to tell foreground from background.
+
+Image segmentation is a form of supervised learning: Some kind of ground truth is needed. Here, it comes in form of a *mask* -- an image, of spatial resolution identical to that of the input data, that designates the true class for every pixel. Accordingly, classification loss is calculated pixel-wise; losses are then summed up to yield an aggregate to be used in optimization.
+
+The "canonical" architecture for image segmentation, around since 2015, is *U-Net.*
+
+## U-Net
+
+Here is the prototypical U-Net, as depicted in the original Rönneberger et al. paper [@RonnebergerFB15].
+
+Of this architecture, numerous variants exist; you could use different layer sizes, activations, ways to achieve downsizing and upsizing, and more. However, there is one defining characteristic: The U-shape, stabilized by the "bridges" crossing over horizontally at all levels.
+
+![](images/unet.png "The original U-Net, as depicted in Ronnerberger et al. (2015).")
+
+In a nutshell, the left-hand side of the U resembles the convolutional architectures used in image classification: It successively reduces spatial resolution. At the same time, another dimension -- the *channels* dimension -- is used to build up a hierarchy of features, ranging from very basic to very specialized.
+
+Unlike in classification, however, the output should have the same spatial resolution as the input. Thus, we need to upsize again -- this is taken care of by the right-hand side of the U. But: How are we going to arrive at a good *per-pixel* classification, now that so much spatial information has been lost?
+
+This is what the "bridges" are for: At each level, the input to an upsampling layer is a *concatenation* of the previous layer's output -- which went through the whole compression/decompression routine -- and some preserved intermediate representation from the downsizing phase. In this way, a U-Net architecture combines attention to detail with feature extraction.
+
+## Brain image segmentation
+
+With U-Net, domain applicability is as broad as the architecture is flexible. Here, we want to detect abnormalities in brain scans. The dataset, used in @BUDA2019218, contains MR images together with manually created [FLAIR](https://en.wikipedia.org/wiki/Fluid-attenuated_inversion_recovery) abnormality segmentation masks. It is available on [Kaggle](https://www.kaggle.com/mateuszbuda/lgg-mri-segmentation).
```

Comment:
I don't read a lot about MRIs, but I've never come across MR images (with just the magnetic resonance part abbreviated) before. Given you've got FLAIR so close by, maybe just spell it out?

Reply:
will do!